lens/private/list/flatten.scrbl

```diff
@@ -0,0 +1,63 @@
+#lang scribble/manual
+
+@(require "../doc-util/main.rkt")
+
+@title{Flattening and unflattening lists}
+
+@defthing[append*-lens lens?]{
+A lens that flattens a list one-level down when viewing, and restores
+the original structure when setting.
+@lens-unstable-examples[
+  (lens-view append*-lens '((a) (b c) () (d e f)))
+  (lens-set append*-lens '((a) (b c) () (d e f)) '(1 2 3 4 5 6))
+]}
+
+@defproc[(flatten/depth-lens [n exact-nonnegative-integer?]) lens?]{
+Creates a lens that flattens a list of depth @racket[n] when
+viewing, and restores the original structure when setting.
```

Comment:
More prose about special values of `n` would be useful, like "A flat list is a list of depth 1, and a single element is a list of depth 0`. It's important to include prose and examples for those cases too.